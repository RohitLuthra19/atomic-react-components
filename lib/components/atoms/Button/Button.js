// @flow
import * as React from 'react';
import styled from 'styled-components';

import styles from './Button.style';

type Props = {
  className: string,
  children: Node,
  inheritedStyles?: string,
  type?: string,
  primary?: boolean,
  secondary?: boolean,
  elementRef?: any => void,
  disabled?: boolean,
  ariaLabel?: string,
};

const Button = ({
  className,
  children,
  inheritedStyles,
  type,
  primary,
  secondary,
  elementRef,
  disabled,
  ariaLabel,
  ...others
}: Props): Node => (
  <button
    aria-label={ariaLabel || null}
    className={className}
    disabled={disabled}
    ref={elementRef}
    type={type}
    {...others}
  >
    {children}
  </button>
);

Button.defaultProps = {
  inheritedStyles: '',
  type: 'button',
  primary: false,
  secondary: false,
  elementRef: undefined,
  disabled: false,
  ariaLabel: '',
};

export default styled(Button)`${styles}`;
export { Button as ButtonVanilla };
