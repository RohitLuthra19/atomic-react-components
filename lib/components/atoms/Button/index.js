// @flow

export { default } from './Button';
export { ButtonVanilla } from './Button';
export type { Props } from './Button';
