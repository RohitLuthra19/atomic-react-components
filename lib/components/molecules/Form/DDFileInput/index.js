// @flow

export { default } from './DDFileInput';
