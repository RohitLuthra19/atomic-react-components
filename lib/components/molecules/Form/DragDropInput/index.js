// @flow

export { default } from './DragDropInput';
