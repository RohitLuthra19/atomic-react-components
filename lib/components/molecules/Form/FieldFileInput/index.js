// @flow

export { default } from './FieldFileInput';
