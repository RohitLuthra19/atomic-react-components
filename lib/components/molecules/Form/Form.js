// @flow
import React from 'react';
import { Formik, Form as FormWrapper } from 'formik';
import type { Node } from 'react';
import Debug from './Debug';
import Input from './Input';
import Select from './Select';


type Props = {
  className?: string,
  action: string,
  initialValues ?: any,
  children: Node,
  onSubmit: Function,
  debug ?: boolean,
  validationSchema ?: Object,
  validate ?: Function
};

class Form extends React.PureComponent<Props> {
  static defaultProps = {
    debug: false,
    className: '',
    initialValues: {},
    validationSchema: {},
    validate: () => false,
  };

  static Input = Input;

  static Select = Select;

  render() {
    const {
      action,
      className,
      onSubmit,
      initialValues,
      children,
      debug,
      validationSchema,
      validate,
      ...others
    } = this.props;

    return (
      <Formik
        initialValues={initialValues}
        validationSchema={validationSchema}
        validate={validate}
        onSubmit={(values) => {
          onSubmit(values);
        }}
        render={() => (
          <FormWrapper {...others}>
            {children}
            {debug && <Debug />}
          </FormWrapper>
        )}
      />
    );
  }
}

export default Form;
