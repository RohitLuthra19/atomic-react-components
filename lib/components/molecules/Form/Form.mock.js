import { action } from '@storybook/addon-actions';

const userFormProps = {
  initialValues: {
    email: '',
    username: '',
    typeOfUser: '',
  },
  debug: true,
  action: '/',
  validationSchema: {
    username: {
      max: [15, 'Max limit reached'],
      required: 'User name Required',
    },
    email: {
      max: [30, 'Max reached'],
      email: 'Invalid email format',
      required: 'Required',
    },
  },
  handleSubmit: action('On Submit Fired'),
};

const userFormSelectProps = {
  options: ['Admin', 'Read Only'],
  name: 'typeOfUser',
  label: 'Type of user',
};

const userFormNameInputProps = {
  name: 'username',
  type: 'text',
  label: 'User Name',
};

const userFormEmailProps = {
  type: 'email',
  name: 'email',
  label: 'Email',
};

export { userFormProps, userFormSelectProps, userFormNameInputProps, userFormEmailProps };
