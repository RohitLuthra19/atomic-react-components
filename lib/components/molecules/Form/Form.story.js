// @flow
import React from 'react';
import { storiesOf } from '@storybook/react';
import {
  userFormProps,
  userFormSelectProps,
  userFormNameInputProps,
  userFormEmailProps,
  userFormAgeGroupProps,
  userFormTnCProps,
} from './Form.mock';

// Import Vanilla Component to display markup and props
import Form from './Form';
import Button from '../../atoms/Button';
// Import Styled Component to showcase variations

storiesOf('Molecules', module).addWithChapters('Form', {
  chapters: [
    {
      sections: [
        {
          sectionFn: () => (
            <Form {...userFormProps}>
              <div>
                <Form.Select {...userFormSelectProps} />
                <Form.Input {...userFormNameInputProps} />
                <Form.Input {...userFormEmailProps} />
                <Form.RadioGroup {...userFormAgeGroupProps} />
                <Form.Checkbox {...userFormTnCProps} />
              </div>
              <Button primary type="submit">Submit</Button>
            </Form>
          ),
          options: {
            showSource: true,
            allowSourceToggling: true,
            showPropTables: true,
            allowPropTablesToggling: true,
          },
        },
      ],
    },
  ],
});
