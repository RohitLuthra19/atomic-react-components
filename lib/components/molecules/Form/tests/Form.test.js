import React from 'react';
import * as Yup from 'yup';
import toJson from 'enzyme-to-json';
import { shallowWithTheme, mountWithTheme } from '../../../../../__mock__/themeMock';
import Form from '..';
import { ButtonVanilla } from '../../../atoms/Button';

describe('Form Component', () => {
  const userFormProps = {
    initialValues: {
      email: '',
      username: '',
      typeOfUser: '',
    },
    debug: false,
    action: '/',
    validationSchema: {
      username: {
        max: [20, 'Max limit reached'],
        required: 'User name Required',
      },
      email: {
        max: [35, 'Max reached'],
        email: 'Invalid email format',
        required: 'Required',
      },
    },
  };

  const SignupSchema = Yup.object().shape({
    username: Yup.string()
      .min(2, 'Too Short!')
      .max(50, 'Too Long!')
      .required('Required'),
  });

  test('should render select tag correctly', () => {
    const userFormSelectProps = {
      options: ['Admin', 'Read Only'],
      name: 'typeOfUsers',
      label: 'Type of users',
      validateOnBlur: true,
    };
    const SelectForm = (
      <Form {...userFormProps}>
        <div>
          <Form.Select {...userFormSelectProps} />
        </div>
        <ButtonVanilla primary type="submit">
          Submit
        </ButtonVanilla>
      </Form>
    );
    const Wrapper = shallowWithTheme(SelectForm);
    expect(toJson(Wrapper)).toMatchSnapshot();
  });

  test('should render input text field', () => {
    const userFormNameInputProps = {
      name: 'username',
      type: 'text',
      label: 'User Names',
    };
    const InputForm = (
      <Form {...userFormProps}>
        <div>
          <Form.Input {...userFormNameInputProps} />
        </div>
        <ButtonVanilla primary type="submit">
          Submit
        </ButtonVanilla>
      </Form>
    );
    const Wrapper = shallowWithTheme(InputForm);
    expect(toJson(Wrapper)).toMatchSnapshot();
  });

  test('should render select field', () => {
    const userFormSelectProps = {
      options: ['Admin', 'Read Only'],
      name: 'typeOfUsers',
      label: 'Type of users',
    };
    const SelectForm = (
      <Form {...userFormProps}>
        <div>
          <Form.Select {...userFormSelectProps} />
        </div>
        <ButtonVanilla primary type="submit">
          Submit
        </ButtonVanilla>
      </Form>
    );
    const Wrapper = shallowWithTheme(SelectForm);
    expect(toJson(Wrapper)).toMatchSnapshot();
  });

  test('should render input field with yupValidationSchema', () => {
    const userFormNameInputProps = {
      name: 'username',
      type: 'text',
      label: 'User Name',
    };
    const InputForm = (
      <Form {...userFormProps} yupValidationSchema={SignupSchema}>
        <div>
          <Form.Input {...userFormNameInputProps} />
        </div>
        <ButtonVanilla primary type="submit">
          Submit
        </ButtonVanilla>
      </Form>
    );
    const Wrapper = shallowWithTheme(InputForm);
    expect(toJson(Wrapper)).toMatchSnapshot();
  });

  test('should render input field with no validation', () => {
    const userFormNameInputProps = {
      name: 'username',
      type: 'text',
      label: 'User Name',
    };

    const formProps = { ...userFormProps };
    delete formProps.validationSchema;
    const InputForm = (
      <Form {...formProps}>
        <div>
          <Form.Input {...userFormNameInputProps} />
        </div>
        <ButtonVanilla primary type="submit">
          Submit
        </ButtonVanilla>
      </Form>
    );
    const Wrapper = shallowWithTheme(InputForm);
    expect(toJson(Wrapper)).toMatchSnapshot();
  });

  test('getSchema should return null if no params are passed', () => {
    const userFormNameInputProps = {
      name: 'username',
      type: 'text',
      label: 'User Name',
    };

    const formProps = { ...userFormProps };
    delete formProps.validationSchema;
    const InputForm = (
      <Form {...formProps}>
        <div>
          <Form.Input {...userFormNameInputProps} />
        </div>
        <ButtonVanilla primary type="submit">
          Submit
        </ButtonVanilla>
      </Form>
    );
    const Wrapper = mountWithTheme(InputForm);
    expect(Wrapper.instance().getSchema()).toBeNull();
  });
});
