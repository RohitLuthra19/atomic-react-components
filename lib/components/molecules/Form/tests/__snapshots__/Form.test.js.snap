// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Form Component should render input text field 1`] = `
<Form
  action="/"
  className=""
  debug={false}
  formWrapperProps={Object {}}
  initialValues={
    Object {
      "email": "",
      "typeOfUser": "",
      "username": "",
    }
  }
  validate={[Function]}
  validateOnBlur={true}
  validateOnChange={true}
  validationSchema={
    Object {
      "email": Object {
        "email": "Invalid email format",
        "max": Array [
          30,
          "Max reached",
        ],
        "required": "Required",
      },
      "username": Object {
        "max": Array [
          15,
          "Max limit reached",
        ],
        "required": "User name Required",
      },
    }
  }
  yupValidationSchema={null}
>
  <Formik
    enableReinitialize={false}
    initialValues={
      Object {
        "email": "",
        "typeOfUser": "",
        "username": "",
      }
    }
    isInitialValid={false}
    render={[Function]}
    validate={[Function]}
    validateOnBlur={true}
    validateOnChange={true}
    validationSchema={
      ObjectSchema {
        "_blacklist": RefSet {
          "list": Set {},
          "refs": Map {},
        },
        "_conditions": Array [],
        "_defaultDefault": [Function],
        "_deps": Array [],
        "_excludedEdges": Array [],
        "_exclusive": Object {},
        "_mutate": false,
        "_nodes": Array [
          "email",
          "username",
        ],
        "_options": Object {
          "abortEarly": true,
          "recursive": true,
        },
        "_type": "object",
        "_typeError": [Function],
        "_whitelist": RefSet {
          "list": Set {},
          "refs": Map {},
        },
        "fields": Object {
          "email": StringSchema {
            "_blacklist": RefSet {
              "list": Set {},
              "refs": Map {},
            },
            "_conditions": Array [],
            "_deps": Array [],
            "_exclusive": Object {
              "max": true,
              "required": false,
              "undefined": false,
            },
            "_mutate": false,
            "_options": Object {
              "abortEarly": true,
              "recursive": true,
            },
            "_type": "string",
            "_typeError": [Function],
            "_whitelist": RefSet {
              "list": Set {},
              "refs": Map {},
            },
            "tests": Array [
              [Function],
              [Function],
              [Function],
              [Function],
            ],
            "transforms": Array [
              [Function],
            ],
          },
          "username": StringSchema {
            "_blacklist": RefSet {
              "list": Set {},
              "refs": Map {},
            },
            "_conditions": Array [],
            "_deps": Array [],
            "_exclusive": Object {
              "max": true,
              "required": false,
            },
            "_mutate": false,
            "_options": Object {
              "abortEarly": true,
              "recursive": true,
            },
            "_type": "string",
            "_typeError": [Function],
            "_whitelist": RefSet {
              "list": Set {},
              "refs": Map {},
            },
            "tests": Array [
              [Function],
              [Function],
              [Function],
            ],
            "transforms": Array [
              [Function],
            ],
          },
        },
        "tests": Array [],
        "transforms": Array [
          [Function],
        ],
      }
    }
  >
    <Form
      noValidate={true}
    >
      <Component
        formik={
          Object {
            "dirty": false,
            "errors": Object {},
            "handleBlur": [Function],
            "handleChange": [Function],
            "handleReset": [Function],
            "handleSubmit": [Function],
            "initialValues": Object {
              "email": "",
              "typeOfUser": "",
              "username": "",
            },
            "isSubmitting": false,
            "isValid": false,
            "isValidating": false,
            "registerField": [Function],
            "resetForm": [Function],
            "setError": [Function],
            "setErrors": [Function],
            "setFieldError": [Function],
            "setFieldTouched": [Function],
            "setFieldValue": [Function],
            "setFormikState": [Function],
            "setStatus": [Function],
            "setSubmitting": [Function],
            "setTouched": [Function],
            "setValues": [Function],
            "submitCount": 0,
            "submitForm": [Function],
            "touched": Object {},
            "unregisterField": [Function],
            "validate": [Function],
            "validateField": [Function],
            "validateForm": [Function],
            "validateOnBlur": true,
            "validateOnChange": true,
            "validationSchema": ObjectSchema {
              "_blacklist": RefSet {
                "list": Set {},
                "refs": Map {},
              },
              "_conditions": Array [],
              "_defaultDefault": [Function],
              "_deps": Array [],
              "_excludedEdges": Array [],
              "_exclusive": Object {},
              "_mutate": false,
              "_nodes": Array [
                "email",
                "username",
              ],
              "_options": Object {
                "abortEarly": true,
                "recursive": true,
              },
              "_type": "object",
              "_typeError": [Function],
              "_whitelist": RefSet {
                "list": Set {},
                "refs": Map {},
              },
              "fields": Object {
                "email": StringSchema {
                  "_blacklist": RefSet {
                    "list": Set {},
                    "refs": Map {},
                  },
                  "_conditions": Array [],
                  "_deps": Array [],
                  "_exclusive": Object {
                    "max": true,
                    "required": false,
                    "undefined": false,
                  },
                  "_mutate": false,
                  "_options": Object {
                    "abortEarly": true,
                    "recursive": true,
                  },
                  "_type": "string",
                  "_typeError": [Function],
                  "_whitelist": RefSet {
                    "list": Set {},
                    "refs": Map {},
                  },
                  "tests": Array [
                    [Function],
                    [Function],
                    [Function],
                    [Function],
                  ],
                  "transforms": Array [
                    [Function],
                  ],
                },
                "username": StringSchema {
                  "_blacklist": RefSet {
                    "list": Set {},
                    "refs": Map {},
                  },
                  "_conditions": Array [],
                  "_deps": Array [],
                  "_exclusive": Object {
                    "max": true,
                    "required": false,
                  },
                  "_mutate": false,
                  "_options": Object {
                    "abortEarly": true,
                    "recursive": true,
                  },
                  "_type": "string",
                  "_typeError": [Function],
                  "_whitelist": RefSet {
                    "list": Set {},
                    "refs": Map {},
                  },
                  "tests": Array [
                    [Function],
                    [Function],
                    [Function],
                  ],
                  "transforms": Array [
                    [Function],
                  ],
                },
              },
              "tests": Array [],
              "transforms": Array [
                [Function],
              ],
            },
            "values": Object {
              "email": "",
              "typeOfUser": "",
              "username": "",
            },
          }
        }
        noValidate={true}
      >
        <form
          noValidate={true}
          onReset={[Function]}
          onSubmit={[Function]}
        >
          <div>
            <FieldInput
              label="User Name"
              name="username"
              type="text"
            >
              <FieldInput
                className="FieldInput-sc-1a7fz5e-0 htIuCH"
                fieldProps={Object {}}
                label="User Name"
                labelProps={Object {}}
                name="username"
                onBlur={null}
                onChange={null}
                placeholder=""
                type="text"
                value=""
              >
                <div
                  className="FieldInput-sc-1a7fz5e-0 htIuCH"
                >
                  <Label
                    className=""
                  >
                    <label
                      className=""
                    >
                      User Name
                    </label>
                  </Label>
                  <C
                    className="ank"
                    name="username"
                    render={[Function]}
                    type="text"
                  >
                    <FieldInner
                      className="ank"
                      formik={
                        Object {
                          "dirty": false,
                          "errors": Object {},
                          "handleBlur": [Function],
                          "handleChange": [Function],
                          "handleReset": [Function],
                          "handleSubmit": [Function],
                          "initialValues": Object {
                            "email": "",
                            "typeOfUser": "",
                            "username": "",
                          },
                          "isSubmitting": false,
                          "isValid": false,
                          "isValidating": false,
                          "registerField": [Function],
                          "resetForm": [Function],
                          "setError": [Function],
                          "setErrors": [Function],
                          "setFieldError": [Function],
                          "setFieldTouched": [Function],
                          "setFieldValue": [Function],
                          "setFormikState": [Function],
                          "setStatus": [Function],
                          "setSubmitting": [Function],
                          "setTouched": [Function],
                          "setValues": [Function],
                          "submitCount": 0,
                          "submitForm": [Function],
                          "touched": Object {},
                          "unregisterField": [Function],
                          "validate": [Function],
                          "validateField": [Function],
                          "validateForm": [Function],
                          "validateOnBlur": true,
                          "validateOnChange": true,
                          "validationSchema": ObjectSchema {
                            "_blacklist": RefSet {
                              "list": Set {},
                              "refs": Map {},
                            },
                            "_conditions": Array [],
                            "_defaultDefault": [Function],
                            "_deps": Array [],
                            "_excludedEdges": Array [],
                            "_exclusive": Object {},
                            "_mutate": false,
                            "_nodes": Array [
                              "email",
                              "username",
                            ],
                            "_options": Object {
                              "abortEarly": true,
                              "recursive": true,
                            },
                            "_type": "object",
                            "_typeError": [Function],
                            "_whitelist": RefSet {
                              "list": Set {},
                              "refs": Map {},
                            },
                            "fields": Object {
                              "email": StringSchema {
                                "_blacklist": RefSet {
                                  "list": Set {},
                                  "refs": Map {},
                                },
                                "_conditions": Array [],
                                "_deps": Array [],
                                "_exclusive": Object {
                                  "max": true,
                                  "required": false,
                                  "undefined": false,
                                },
                                "_mutate": false,
                                "_options": Object {
                                  "abortEarly": true,
                                  "recursive": true,
                                },
                                "_type": "string",
                                "_typeError": [Function],
                                "_whitelist": RefSet {
                                  "list": Set {},
                                  "refs": Map {},
                                },
                                "tests": Array [
                                  [Function],
                                  [Function],
                                  [Function],
                                  [Function],
                                ],
                                "transforms": Array [
                                  [Function],
                                ],
                              },
                              "username": StringSchema {
                                "_blacklist": RefSet {
                                  "list": Set {},
                                  "refs": Map {},
                                },
                                "_conditions": Array [],
                                "_deps": Array [],
                                "_exclusive": Object {
                                  "max": true,
                                  "required": false,
                                },
                                "_mutate": false,
                                "_options": Object {
                                  "abortEarly": true,
                                  "recursive": true,
                                },
                                "_type": "string",
                                "_typeError": [Function],
                                "_whitelist": RefSet {
                                  "list": Set {},
                                  "refs": Map {},
                                },
                                "tests": Array [
                                  [Function],
                                  [Function],
                                  [Function],
                                ],
                                "transforms": Array [
                                  [Function],
                                ],
                              },
                            },
                            "tests": Array [],
                            "transforms": Array [
                              [Function],
                            ],
                          },
                          "values": Object {
                            "email": "",
                            "typeOfUser": "",
                            "username": "",
                          },
                        }
                      }
                      name="username"
                      render={[Function]}
                      type="text"
                    >
                      <Input
                        name="username"
                        onBlur={[Function]}
                        onChange={[Function]}
                        placeholder=""
                        type="text"
                        value=""
                      >
                        <Input
                          ariaLabel=""
                          className="Input-sc-1gq1l3u-0 dkCeGC"
                          name="username"
                          onBlur={[Function]}
                          onChange={[Function]}
                          placeholder=""
                          type="text"
                          value=""
                        >
                          <input
                            aria-label=""
                            className="Input-sc-1gq1l3u-0 dkCeGC"
                            name="username"
                            onBlur={[Function]}
                            onChange={[Function]}
                            placeholder=""
                            type="text"
                            value=""
                          />
                        </Input>
                      </Input>
                    </FieldInner>
                  </C>
                  <C
                    component={[Function]}
                    name="username"
                  >
                    <ErrorMessageImpl
                      component={[Function]}
                      formik={
                        Object {
                          "dirty": false,
                          "errors": Object {},
                          "handleBlur": [Function],
                          "handleChange": [Function],
                          "handleReset": [Function],
                          "handleSubmit": [Function],
                          "initialValues": Object {
                            "email": "",
                            "typeOfUser": "",
                            "username": "",
                          },
                          "isSubmitting": false,
                          "isValid": false,
                          "isValidating": false,
                          "registerField": [Function],
                          "resetForm": [Function],
                          "setError": [Function],
                          "setErrors": [Function],
                          "setFieldError": [Function],
                          "setFieldTouched": [Function],
                          "setFieldValue": [Function],
                          "setFormikState": [Function],
                          "setStatus": [Function],
                          "setSubmitting": [Function],
                          "setTouched": [Function],
                          "setValues": [Function],
                          "submitCount": 0,
                          "submitForm": [Function],
                          "touched": Object {},
                          "unregisterField": [Function],
                          "validate": [Function],
                          "validateField": [Function],
                          "validateForm": [Function],
                          "validateOnBlur": true,
                          "validateOnChange": true,
                          "validationSchema": ObjectSchema {
                            "_blacklist": RefSet {
                              "list": Set {},
                              "refs": Map {},
                            },
                            "_conditions": Array [],
                            "_defaultDefault": [Function],
                            "_deps": Array [],
                            "_excludedEdges": Array [],
                            "_exclusive": Object {},
                            "_mutate": false,
                            "_nodes": Array [
                              "email",
                              "username",
                            ],
                            "_options": Object {
                              "abortEarly": true,
                              "recursive": true,
                            },
                            "_type": "object",
                            "_typeError": [Function],
                            "_whitelist": RefSet {
                              "list": Set {},
                              "refs": Map {},
                            },
                            "fields": Object {
                              "email": StringSchema {
                                "_blacklist": RefSet {
                                  "list": Set {},
                                  "refs": Map {},
                                },
                                "_conditions": Array [],
                                "_deps": Array [],
                                "_exclusive": Object {
                                  "max": true,
                                  "required": false,
                                  "undefined": false,
                                },
                                "_mutate": false,
                                "_options": Object {
                                  "abortEarly": true,
                                  "recursive": true,
                                },
                                "_type": "string",
                                "_typeError": [Function],
                                "_whitelist": RefSet {
                                  "list": Set {},
                                  "refs": Map {},
                                },
                                "tests": Array [
                                  [Function],
                                  [Function],
                                  [Function],
                                  [Function],
                                ],
                                "transforms": Array [
                                  [Function],
                                ],
                              },
                              "username": StringSchema {
                                "_blacklist": RefSet {
                                  "list": Set {},
                                  "refs": Map {},
                                },
                                "_conditions": Array [],
                                "_deps": Array [],
                                "_exclusive": Object {
                                  "max": true,
                                  "required": false,
                                },
                                "_mutate": false,
                                "_options": Object {
                                  "abortEarly": true,
                                  "recursive": true,
                                },
                                "_type": "string",
                                "_typeError": [Function],
                                "_whitelist": RefSet {
                                  "list": Set {},
                                  "refs": Map {},
                                },
                                "tests": Array [
                                  [Function],
                                  [Function],
                                  [Function],
                                ],
                                "transforms": Array [
                                  [Function],
                                ],
                              },
                            },
                            "tests": Array [],
                            "transforms": Array [
                              [Function],
                            ],
                          },
                          "values": Object {
                            "email": "",
                            "typeOfUser": "",
                            "username": "",
                          },
                        }
                      }
                      name="username"
                    />
                  </C>
                </div>
              </FieldInput>
            </FieldInput>
          </div>
          <Button
            primary={true}
            type="submit"
          >
            <Button
              ariaLabel=""
              className="Button-w5odot-0 fDsdHk"
              disabled={false}
              inheritedStyles=""
              primary={true}
              secondary={false}
              type="submit"
            >
              <button
                aria-label={null}
                className="Button-w5odot-0 fDsdHk"
                disabled={false}
                type="submit"
              >
                Submit
              </button>
            </Button>
          </Button>
        </form>
      </Component>
    </Form>
  </Formik>
</Form>
`;
